src/tcp_server.rs

```diff
@@ -196,6 +196,22 @@ impl TcpServer {
                                                         }
                                                     }
                                                 }
+                                                ClientMessage::RequestCurrentLayerInfo {} => {
+                                                    let mut k = kanata.lock();
+                                                    let cur_layer = k.layout.bm().current_layer();
+                                                    let msg = ServerMessage::CurrentLayerInfo {
+                                                        name: k.layer_info[cur_layer].name.clone(),
+                                                        cfg_text: k.layer_info[cur_layer]
+                                                            .cfg_text
+                                                            .clone(),
+                                                    };
```

Comment:
It would be good to add an explicit `drop(k);` here to release the mutex before writing to the TCP stream, since `k` is no longer needed.